returnn/tf/layers/signal_processing.py

```diff
@@ -681,10 +737,15 @@ def __init__(self, nr_of_channels=1, **kwargs):
 
     input_placeholder = self.input_data.get_placeholder_as_batch_major()
 
-    output = tf.reshape(input_placeholder, [tf.shape(input_placeholder)[0], tf.shape(input_placeholder)[1], nr_of_channels, tf.shape(input_placeholder)[2] / nr_of_channels])
-    self.output.placeholder = tf.transpose(tf.reshape(tf.transpose(output, [1, 3, 0, 2]), (tf.shape(output)[1], tf.shape(output)[3], tf.shape(output)[0] * tf.shape(output)[2])), [2, 0, 1])
+    output = tf.reshape(input_placeholder,
+                        [tf.shape(input_placeholder)[0], tf.shape(input_placeholder)[1], nr_of_channels,
+                         tf.shape(input_placeholder)[2] / nr_of_channels])
+    self.output.placeholder = tf.transpose(tf.reshape(tf.transpose(output, [1, 3, 0, 2]), (
+      tf.shape(output)[1], tf.shape(output)[3], tf.shape(output)[0] * tf.shape(output)[2])), [2, 0, 1])
```

Comment:
I would prefer such code to be multiple separate statements, where you also comment on the shape after each transformation. This is too complicated to read otherwise.

Reply:
I fully agree. However, I don't have any setup which uses this code so I cannot test it. If I change stuff here, I would have to assume how it was used (e.g. input shapes) and hope that it does not affect the behaviour.

Reply:
This doesn't need testing. This would just be a straight-forward code transformation. This is not different to what you do already now.
About the comments, also this doesn't need testing. You can simply read that from the code. I just mean to write it like [B,T,D], or whatever there is.